src/main/java/org/opensearch/flowframework/workflow/GetMLTaskStep.java

```diff
@@ -95,4 +82,42 @@ public String getName() {
         return NAME;
     }
 
+    private void retryableGetMlTask(List<WorkflowData> data, CompletableFuture<WorkflowData> getMLTaskFuture, String taskId, int retries) {
+        mlClient.getTask(taskId, ActionListener.wrap(response -> {
+            if (response.getState() != MLTaskState.COMPLETED) {
+                throw new IllegalStateException("MLTask is not yet completed");
+            } else {
+                logger.info("ML Task retrieval successful");
+                getMLTaskFuture.complete(
+                    new WorkflowData(
+                        Map.ofEntries(
+                            Map.entry(MODEL_ID, response.getModelId()),
+                            Map.entry(REGISTER_MODEL_STATUS, response.getState().name())
+                        ),
+                        data.get(0).getWorkflowId()
+                    )
+                );
+            }
+        }, exception -> {
+            if (shouldRetry(getMLTaskFuture, retries)) {
+                final int retryAdd = retries + 1;
+                retryableGetMlTask(data, getMLTaskFuture, taskId, retryAdd);
+            } else {
+                logger.error("Failed to retrieve ML Task, maximum retries exceeded");
+                getMLTaskFuture.completeExceptionally(
+                    new FlowFrameworkException(exception.getMessage(), ExceptionsHelper.status(exception))
+                );
+            }
+        }));
+    }
+
+    private boolean shouldRetry(CompletableFuture<WorkflowData> getMLTaskFuture, int retries) {
+        try {
+            Thread.sleep(5000);
+        } catch (Exception e) {
+            getMLTaskFuture.completeExceptionally(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+        }
+        return retries < maxRequestRetrySetting.getMaxRetries();
+    }
```

Comment:
This method should be in a Util Class. We should make it generic.

Reply:
I'm somewhat against making this a utility method since wait times between requests really depends on the API that the workflow step is invoking. In this case, we sleep the thread for about 5 seconds between retries since the local model registration task takes about 30 seconds. It may be different for other cases.

Reply:
Let's take the sleep time as a param then :)

Reply:
will do :)

Reply:
On second thought, `shouldRetry` shouldn't be responsible for waiting before proceeding with another request, it should just indicate whether or not a retry should occur. I'll go ahead and add that utility class but I'll omit the thread sleep from this

Reply:
Agreed!